File: src/DynamicLinkLibraries/3DConverters/Abstract3DConverters/Materials/SimpleMaterial.cs

namespace Abstract3DConverters.Materials
{
    public abstract class SimpleMaterial : Material
    {
        public Color Color { get; private set; }

        protected SimpleMaterial(Color color)
        {
            if (color != null)
            {
                Color = color.Clone() as Color;
            }
        }

    }
}

File: src/Web/reactprojectsimplegame/src/Library/Abstract3DConverters/Materials/SimpleMaterial.ts
import { ColorTexture } from "../ColorTexture";
import { ImageTexture } from "../ImageTexture";
import { Material } from "./Material";

export class SimpleMaterial extends Material {

    constructor(name : string, color: ColorTexture) {
        super(name)
        this.color = color
        this.types.push("SimpleMaterial")
        this.typeName = "SimpleMaterial"
    }
    protected color: ColorTexture = new ColorTexture([])

    protected image !: ImageTexture


}